`pkgsCross.mingwW64.libusb1`: build fails with `nix-build`, succeeds with `nix-shell`

### Steps To Reproduce
Steps to reproduce the behavior:
```
$ nix-build '<nixpkgs>' -A pkgsCross.mingwW64.libusb1
[...]
make  all-recursive
make[1]: Entering directory '/build/source'
Making all in libusb
make[2]: Entering directory '/build/source/libusb'
  CC       core.lo
  CC       descriptor.lo
  CC       hotplug.lo
  CC       io.lo
  CC       strerror.lo
  CC       sync.lo
  CC       os/events_windows.lo
  CC       os/threads_windows.lo
  RC       libusb-1.0.lo
libusb-1.0.rc:11: fatal error: opening dependency file .deps/libusb-1.0.Tpo: No such file or directory
   11 | #ifndef LIBUSB_VERSIONSTRING
      |
compilation terminated.
x86_64-w64-mingw32-windres: preprocessing failed.
make[2]: *** [Makefile:982: libusb-1.0.lo] Error 1
make[2]: Leaving directory '/build/source/libusb'
make[1]: *** [Makefile:490: all-recursive] Error 1
make[1]: Leaving directory '/build/source'
make: *** [Makefile:395: all] Error 2
```

So when I wanted to dig deeper, with
```
$ nix-shell '<nixpkgs>' -A pkgsCross.mingwW64.libusb1 --run genericBuild
[...]
make  all-recursive
make[1]: Entering directory '/tmp/source'
Making all in libusb
make[2]: Entering directory '/tmp/source/libusb'
  CC       core.lo
  CC       descriptor.lo
  CC       hotplug.lo
  CC       io.lo
  CC       strerror.lo
  CC       sync.lo
  CC       os/events_windows.lo
  CC       os/threads_windows.lo
  RC       libusb-1.0.lo
  CC       os/windows_common.lo
  CC       os/windows_usbdk.lo
  CC       os/windows_winusb.lo
[fine apart from warnings until install]
```

### Build log
[nix-shell-mingwW64-libusb.log](https://github.com/NixOS/nixpkgs/files/9212908/nix-shell-mingwW64-libusb.log)
[nix-build-mingwW64-libusb.log](https://github.com/NixOS/nixpkgs/files/9212915/nix-build-mingwW64-libusb.log)


### Additional context
I'm trying to recreate https://github.com/DavidEGrayson/nixcrpkgs/tree/master/pkgs/openocd with a toolchain that's on hydra so I don't have to compile GCC for myself (twice!)

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.18.12, NixOS, 22.11 (Raccoon), 22.11.20220718.9eb60f2`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.10.3`
 - channels(root): `""`
 - nixpkgs: `/nix/store/4la4fc5wfn4p0784i12h0a046yqzln16-source`
```
In particular, `nixpkgs` is pinned to:
```console
[user@system:~]$ nixos-version --json | jq
{
  "configurationRevision": "e142ae2a9fb132ef42f7b9799d5fa7ed8aee398e",
  "nixosVersion": "22.11.20220718.9eb60f2",
  "nixpkgsRevision": "9eb60f25aff0d2218c848dd4574a0ab5e296cabe"
}
```

--- a/pkgs/development/libraries/libusb1/default.nix
+++ b/pkgs/development/libraries/libusb1/default.nix
@@ -33,6 +33,9 @@ stdenv.mkDerivation rec {
 
   dontDisableStatic = withStatic;
 
+  # libusb-1.0.rc:11: fatal error: opening dependency file .deps/libusb-1.0.Tpo: No such file or directory
+  dontAddDisableDepTrack = stdenv.hostPlatform.isWindows;
+
   configureFlags =
     lib.optional (!enableUdev) "--disable-udev"
     ++ lib.optional (withExamples) "--enable-examples-build";

